File: apps/render-ios-playground/render-ios-playground/Infrastructure/DependencyInjection/DIContainer.swift
import Foundation
import Supabase

/// Dependency injection container for the application
class DIContainer {
    static let shared = DIContainer()
    
    private init() {}
    
    lazy var networkClient: NetworkClient = {
        return NetworkClient()
    }()
    
    lazy var supabaseClient: SupabaseClient = {
        return SupabaseClient(
            supabaseURL: URL(string: "https://yhfeoztyhuiccuyeghiw.supabase.co")!,
            supabaseKey: "sb_publishable_8fDYhB0k7n_wuAywpua6vQ_JthMjgzA"
        )
    }()
    
    // MARK: - Repositories
    
    lazy var scenarioRepository: ScenarioRepository = {
        return ScenarioRepositoryImpl(
            networkClient: networkClient,
            supabaseClient: supabaseClient
        )
    }()
    
    // MARK: - Use Cases
    
    lazy var fetchScenarioUseCase: FetchScenarioUseCase = {
        return FetchScenarioUseCase(
            scenarioRepository: scenarioRepository
        )
    }()
    
    // MARK: - Application Services
    
    lazy var scenarioService: ScenarioService = {
        return ScenarioService(
            fetchScenarioUseCase: fetchScenarioUseCase
        )
    }()
    
    // MARK: SDK
    
    lazy var logger: Logger = {
        let consoleLogger = ConsoleLogger(
            minimumLogLevel: .info,
            isEnabled: true,
            includeTimestamp: true
        )
        
        let fileLogger = FileLogger(
            minimumLogLevel: .debug,
            isEnabled: true,
            includeTimestamp: true
        )
        
        return ComposableLogger(loggers: [consoleLogger, fileLogger])
    }()
    
    lazy var componentRegistry: ComponentRegistry = {
        let registry = ComponentRegistry()
        let renderers: [Renderer] = [
            ViewRenderer(),
            RowRenderer(),
            ColumnRenderer(),
            StackRenderer(),
            TextRenderer(),
            ButtonRenderer(),
            ImageRenderer(),
            CheckboxRenderer(),
            StepperRenderer(),
            RatingRenderer(),
            NavbarRenderer()
        ]
        renderers.forEach { registry.register(renderer: $0) }
        return registry
    }()
    
    // MARK: - Logger Management
    
    private var _logger: Logger?
    
    /// Get the current logger instance
    var currentLogger: Logger {
        return _logger ?? logger
    }
    
    /// Update the logger configuration
    func updateLogger(_ newLogger: Logger) {
        _logger = newLogger
    }
}

File: apps/render-android-playground/app/src/main/java/com/max_mrtnv/render_android_playground/infrastructure/di/DIContainer.kt
package com.max_mrtnv.render_android_playground.infrastructure.di

import com.max_mrtnv.render_android_playground.application.services.ScenarioService
import com.max_mrtnv.render_android_playground.application.usecases.FetchScenarioUseCase
import com.max_mrtnv.render_android_playground.application.usecases.RenderComponentUseCase
import com.max_mrtnv.render_android_playground.domain.repositories.ScenarioRepository
import com.max_mrtnv.render_android_playground.domain.services.ComponentRenderingService
import com.max_mrtnv.render_android_playground.domain.services.Renderer
import com.max_mrtnv.render_android_playground.infrastructure.repositories.HttpScenarioRepository
import com.max_mrtnv.render_android_playground.presentation.renderers.ButtonRenderer
import com.max_mrtnv.render_android_playground.presentation.renderers.ContainerRenderer
import com.max_mrtnv.render_android_playground.presentation.renderers.InputRenderer
import com.max_mrtnv.render_android_playground.presentation.renderers.TextRenderer

/**
 * Dependency injection container for the application
 */
object DIContainer {
    
    // Repositories
    val scenarioRepository: ScenarioRepository by lazy {
        HttpScenarioRepository()
    }
    
    // Renderers
    private val renderers: Map<String, Renderer> by lazy {
        mapOf(
            "container" to ContainerRenderer(),
            "button" to ButtonRenderer(),
            "text" to TextRenderer(),
            "input" to InputRenderer()
        )
    }
    
    // Domain Services
    val componentRenderingService: ComponentRenderingService by lazy {
        ComponentRenderingService(renderers)
    }
    
    // Use Cases
    val fetchScenarioUseCase: FetchScenarioUseCase by lazy {
        FetchScenarioUseCase(scenarioRepository)
    }
    
    val renderComponentUseCase: RenderComponentUseCase by lazy {
        RenderComponentUseCase(componentRenderingService)
    }
    
    // Application Services
    val scenarioService: ScenarioService by lazy {
        ScenarioService(fetchScenarioUseCase, renderComponentUseCase)
    }
}
